ip: Support configuring per-device IPv4 sysctl forwarding

cathay4t opened the pull request:
This feature introduces the ability to configure
`net.ipv4.conf.<interface>.forwarding` through nmstate.

Example YAML:

```yml
---
interfaces:
  - name: eth1
    type: ethernet
    state: up
    ipv4:
      address:
        - ip: 192.0.2.251
          prefix-length: 24
      dhcp: false
      enabled: true
      forwarding: true
```

Unit and integration test cases included.

Resolves: https://issues.redhat.com/browse/RHEL-36429

Changes by Gris Ge:
 * Use `version-info` D-BUS API for checking support status of IPv4
   forwarding in NetworkManager instead of comparing version number.
 * Add integration test case for enabled ipv4 forwarding on auto
   interface without DHCP server.

ihuguet commented on rust/src/lib/ip.rs:
```diff
@@ -147,6 +149,12 @@ pub struct InterfaceIpv4 {
+    /// When applying with `Some(true)`, nmstate will enable forwarding.
+    /// When applying with `Some(false)`, nmstate will disable forwarding.
+    /// When applying with `None`, the existing sysctl forwarding setting
+    /// is left untouched.
```
Only the first line of this comment is needed. The rest only describes standard nmstate behavior.

cathay4t replied:
Fixed. Thanks!

ihuguet commented on rust/src/lib/nm/query_apply/apply.rs:
```diff
@@ -392,10 +399,13 @@ fn gen_nm_conn_need_to_deactivate_first(
     let version = if let Ok(ver_info) = nm_api.version_info().await {
         *route_remove_needs_deactivate = !ver_info
             .has_capability(NmVersionInfo::CAPABILITY_SYNC_ROUTE_WITH_TABLE);
+        *ipv4_forward_support =
+            ver_info.has_capability(NmVersionInfo::CAPABILITY_IP4_FORWARDING);
```
Add a `*ipv4_forward_support = false` in the `else` block.

cathay4t replied:
Fixed. Thanks!

ihuguet commented on tests/integration/static_ip_address_test.py:
```diff
@@ -917,6 +924,78 @@ def test_merge_ip_enabled_property_from_current(setup_eth1_static_ip):
+    with pytest.raises(libnmstate.error.NmstateVerificationError):
+        apply_with_description(
+            "Configure the ethernet device eth1 with IPv4 sysctl forwarding "
+            "enabled",
```
disabled

cathay4t replied:
Fixed. Thanks!

liangwen12year committed:
ip: Support configuring per-device IPv4 sysctl forwarding

This feature introduces the ability to configure
`net.ipv4.conf.<interface>.forwarding` through nmstate.

Example YAML:

```yml
---
interfaces:
  - name: eth1
    type: ethernet
    state: up
    ipv4:
      address:
        - ip: 192.0.2.251
          prefix-length: 24
      dhcp: false
      enabled: true
      forwarding: true
```

Unit and integration test cases included.

Resolves: https://issues.redhat.com/browse/RHEL-36429

Changes by Gris Ge:
 * Use `version-info` D-BUS API for checking support status of IPv4
   forwarding in NetworkManager instead of comparing version number.
 * Add integration test case for enabled ipv4 forwarding on auto
   interface without DHCP server.

Signed-off-by: Wen Liang <liangwen12year@gmail.com>
Signed-off-by: Gris Ge <fge@redhat.com>
